src/firewall/core/rich.py

```diff
@@ -661,10 +678,18 @@ def check(self):
             if self.element.protocol not in [ "tcp", "udp", "sctp", "dccp" ]:
                 raise FirewallError(errors.INVALID_PROTOCOL, self.element.protocol)
 
-        # other element and not empty?
-        elif self.element is not None:
-            raise FirewallError(errors.INVALID_RULE, "Unknown element %s" % 
-                                type(self.element))
```

Comment:
Still some whitespace errors here. That's why the diff is showing this section as deleted and then re-added. Looks trailing spaces or tabs.

Reply:
Still unnecessary white space changes here.